packages/d/duckdb/xmake.lua

```diff
@@ -0,0 +1,32 @@
+package("duckdb")
+    set_homepage("http://duckdb.org/")
+    set_description("DuckDB is an in-process SQL OLAP Database Management System")
+    set_license("MIT")
+
+    add_urls("https://github.com/duckdb/duckdb/releases/download/$(version)/libduckdb-src.zip",
+             "https://github.com/duckdb/duckdb.git")
+    add_versions("v0.10.0", "385e27aa67712813e4a07389465c4c5c45c431d97cddd35713b8a306d2a86f2d")
+
+    on_install("macosx", "linux", "windows", function (package)
+        io.writefile("xmake.lua", [[
+            add_rules("mode.debug", "mode.release")
+            set_languages("c++17")
+            target("duckdb")
+                set_kind("shared")
```

Comment:
why not let package decide kind?

Reply:
Because the libduckdb package only contains three file: duckdb.hpp, duckdb.h, duckdb.cpp. If we did not set this, the xmake will complain cannot find main.

Reply:
use `set_kind("$(kind)")`, we need to switch static/shared dynamically.

Reply:
done